src/test_utils/es/es_test_config.js

```diff
@@ -20,13 +20,40 @@ export const esTestConfig = new class EsTestConfig {
   }
 
   getUrlParts() {
+
+    let testEsUrl;
+    let protocol;
+    let hostname;
+    let port;
+    let username;
+    let password;
+
+    // Allow setting one complete TEST_ES_URL for Es like https://elastic:changeme@myCloudInstance:9200
+    if (process.env.TEST_ES_URL) {
+      testEsUrl = url.parse(process.env.TEST_ES_URL);
+      // have to remove the ":" off protocol
+      protocol = testEsUrl.protocol.slice(0, -1);
+      hostname = testEsUrl.hostname;
+      port = parseInt(testEsUrl.port, 10);
+      username = testEsUrl.username;
+      password = testEsUrl.password;
+    } else {
+      // Allow setting any individual component(s) of the URL,
+      // or use default values (username and password from shield.js)
+      protocol = process.env.TEST_ES_PROTOCOL || 'http';
```

Comment:
Do you still need/use the other TEST_ES_* environment variables? Can we get rid of them if we are now providing the full URL?

Reply:
I didn't want to break backward compatibility if anybody was just overriding a port or something like that.  I don't know if anybody is using any of them, but they were documented.